### PR TITLE
cbindgen: new package plus 1 dependency package

### PR DESCRIPTION
**Pull request by teaguesterling**

### CODE REVIEW

File: var/spack/repos/builtin/packages/cbindgen/package.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#    depends_on("rust-log@0.4:")
+#    depends_on("rust-serde@1.0.103+derive")
+#    depends_on("rust-serde_json@1.0:")
+#    depends_on("rust-tempfile@3:")
+#    depends_on("rust-toml@0.8.8:")
+#    depends_on("rust-proc-macro2@1.0.60")
+#    depends_on("rust-quote@1:")
+#    depends_on("rust-heck@0.4:")
+#
+#    depends_on("rust-clap@4.3:")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/cbindgen/package.py]: Not clear why these dependencies are here (and commented out).  Perhaps a comment *if* retaining them.

Are these lines intended to be placeholders for when the packages are added, for example?

Do all of these versions successfully build (without them)?

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and version sha256s but confused about the commented out dependencies.

**Comment by teaguesterling** [comment]
Whoops! Thanks for catching that. Looks like this one wasn't really ready to go. Will review and fix.

**Review by tldahlgren — DISMISSED** [review]
Confirmed homepages, version sha256s, and commits.

**Comment by tldahlgren** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/rust-indexmap/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class RustIndexmap(CargoPackage):
+    """A hash table with consistent order and fast iteration; access items by key or sequence index"""
+
+    # FIXME: Add a proper url for your package's homepage here.
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rust-indexmap/package.py]: Removing this will make the audit check happy.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
See comment

File: var/spack/repos/builtin/packages/rust-indexmap/package.py (hunk 2 above)
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/rust-indexmap/package.py]: :facepalm:

**Comment by teaguesterling** [comment]
@spackbot fix style

**Comment by tldahlgren** [comment]
Please resolve the `flake8` issues:

```
...
  black checks were clean
==> Running flake8 checks
var/spack/repos/builtin/packages/rust-indexmap/package.py:9: [E501] line too long (102 > 99 characters)
var/spack/repos/builtin/packages/rust-proc-macro2/package.py:9: [E501] line too long (142 > 99 characters)
  flake8 found errors
...
```

**Comment by teaguesterling** [comment]
I'm going to move this back to draft unless we want to clean it up and include as-is. There are a lot of rust dependencies that are not yet in Spack and it's a bit of a rabbit hole (see https://github.com/spack/spack/discussions/45483).

Happy to prep this for merging if you agree (to enable Firefox, for example). I wanted to be transparent about the lingering unadressed dependencies in the Rust ecosystem.

**Comment by tldahlgren** [comment]
> I'm going to move this back to draft unless we want to clean it up and include as-is. There are a lot of rust dependencies that are not yet in Spack and it's a bit of a rabbit hole (see #45483).
> 
> Happy to prep this for merging if you agree (to enable Firefox, for example). I wanted to be transparent about the lingering unadressed dependencies in the Rust ecosystem.

As long as it doesn't break other builds, it's up to you.  Just let me know what you want to do.

**Comment by teaguesterling** [comment]
> > I'm going to move this back to draft unless we want to clean it up and include as-is. There are a lot of rust dependencies that are not yet in Spack and it's a bit of a rabbit hole (see #45483).
> > Happy to prep this for merging if you agree (to enable Firefox, for example). I wanted to be transparent about the lingering unadressed dependencies in the Rust ecosystem.
> 
> As long as it doesn't break other builds, it's up to you. Just let me know what you want to do.

I've removed some of the dependencies here. After poking around a bit I think there's more work needed to properly model the rust library dependencies in Spack.

**Review by bernhardkaindl — APPROVED** [review]
CI passes, Built on Ubuntu 22.04:

@tldahlgren, can you retract your "changes requested" status?

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**